### PR TITLE
Add anonymization email and background task

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Adds email to notify users who have not logged in for a while that their account data will be removed soon
- Adds a background task that sends the email to the appropriate users

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3591](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3591)

[TILA-3591]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3591?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tilavarauspalvelu/models/user/queryset.py [hunk 1]
```diff
@@ -80,5 +82,17 @@ def anonymize_inactive_users(self) -> None:
             if user.actions.can_anonymize():
                 user.actions.anonymize()
 
+    def should_anonymize_users(self) -> Self:
```
Comment (vergama) [inline tilavarauspalvelu/models/user/queryset.py:85]: Same as the permission deactivation email: it is desirable to notify a user only once, if fresh login is not performed between the condition first triggering for the user and the eventual anonymization taking place.

Comment (matti-lamppu) [reply tilavarauspalvelu/models/user/queryset.py]: Added field `sent_email_about_anonymization` to user and then checks for this field when gathering email recipients for the email, as well as an update to the after the emails have been sent.

File: tilavarauspalvelu/models/user/queryset.py [hunk 2]
```diff
@@ -80,5 +82,17 @@ def anonymize_inactive_users(self) -> None:
...
+        """Which users should be anonymized due to inactivity (and aren't already anonymized)."""
+        cutoff = (
+            local_date()
+            + datetime.timedelta(days=settings.ANONYMIZATION_NOTIFICATION_BEFORE_DAYS)
+            - datetime.timedelta(days=settings.ANONYMIZE_USER_IF_LAST_LOGIN_IS_OLDER_THAN_DAYS)
+        )
+        return self.exclude(
+            first_name=ANONYMIZED_FIRST_NAME,
+            last_name=ANONYMIZED_LAST_NAME,
+        ).filter(last_login__date__lt=cutoff)
```
[the remark this answers is not in the document]
Comment (matti-lamppu) [reply tilavarauspalvelu/models/user/queryset.py]: Refactored the `UserQuerySet` code to reuse `should_anonymize_users` for this. Did same for `remove_old_superuser_and_staff_permissions` as well.

**Review by vergama — APPROVED** [review]

**Pull request merged**